### ISSUE TITLE
Fluent-bit keeps crashing on windows server core 2019 containers

### ISSUE THREAD

**Issue by Eishi2012** [fluent/fluent-bit#7832]
## Bug Report

**Describe the bug**
fluent-bit keeps crashing on our production setup. We are running fluent-bit v1.9.4 x64 on windows server core 2019 containers. We also tried upgrading fluent-bit to v2.1.8 but the frequency of app crash increased after the upgrade.

**To Reproduce**

- Steps to reproduce the problem:
Run  fluent-bit v1.9.4 x64 on windows server core 2019 containers.

- Example log message if applicable:
```
Fault bucket , type 0
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: fluent-bit.exe
P2: 1.9.4.0
P3: 629d7b07
P4: fluent-bit.exe
P5: 1.9.4.0
P6: 629d7b07
P7: c0000005
P8: 000000000011c434
P9: 
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERE3C9.tmp.dmp
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERE59E.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERE5CE.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERE5CC.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERE5ED.tmp.txt
\\?\C:\Windows\Temp\WERE60F.tmp.appcompat.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportQueue\AppCrash_fluent-bit.exe_ab1fba29d54e95e2bb739186f18343132275d72_0d0720dc_cab_3a8be62a\memory.hdmp

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportQueue\AppCrash_fluent-bit.exe_ab1fba29d54e95e2bb739186f18343132275d72_0d0720dc_cab_3a8be62a

Analysis symbol: 
Rechecking for solution: 0
Report Id: 5bc741ed-4f83-41ce-bb9d-f71de814f168
Report Status: 6
Hashed bucket: 
Cab Guid: 0
```

**Expected behavior**
Service shouldn't crash unexpectedly.

**Your Environment**
* Version used: 1.9.4 and 2.1.8
* Configuration: 
My Config:
```
[SERVICE]
    Flush 5
    Daemon  off
    log_level  Warning
    Parsers_File parsers.conf
    Plugins_File plugins.conf

[INPUT]
    Name  winevtlog
    Channels  Setup,System,Application
    Tag Windows-events
    Interval_Sec  5
    String_Inserts  true

[INPUT]
    Name  tail
    Parser  iis
    Path  C:\\inetpub\\logs\\LogFiles\\*\\u_ex*.log
    Tag  iis

[INPUT]
    Name  tail
    Parser  iis
    Path  C:\\inetpub\\logs\\LogFiles\\*\\u_ex*.log
    Tag  iis_health

[INPUT]
    Name  tail
    Parser  json
    Path  C:\\inetpub\\logs\\application\\*.log
    Buffer_Max_Size 2048k
    Read_from_Head  On
    Tag  application

[INPUT]
    Name  tail
    Parser json
    Path  C:\\inetpub\\logs\\application\\*.ikvm
    Buffer_Max_Size  2048k
    Read_from_Head  On
    Tag  app

# Exclude health check logs
[FILTER]
    Name  grep
    Match iis
    Exclude user_agent ^.*ELB-HealthChecker.*|.*WindowsPowerShell.*$

# Allow logs of failed health checks
[FILTER]
    Name grep
    Match iis_health
    Regex status ^(3|4|5).*$

[FILTER]
    Name aws
    Match *
    imds_version v1
    az true
    ec2_instance_id true
    ec2_instance_type true
    private_ip true
    ami_id true
    account_id true
    vpc_id true

[OUTPUT]
    Name        datadog
    Match       iis*
    Host        http-intake.logs.datadoghq.com
    TLS         on
    compress    gzip
    apikey     ****
    dd_service  ${DD_SERVICE}
    dd_source   iis
    dd_tags     env:${DD_ENV}

[OUTPUT]
    Name        datadog
    Match       app*
    Host        http-intake.logs.datadoghq.com
    TLS         on
    compress    gzip
    apikey     ****
    dd_service  ${DD_SERVICE}
    dd_source   application
    dd_tags     env:${DD_ENV}

[OUTPUT]
    Name        datadog
    Match       Windows-events
    Host        http-intake.logs.datadoghq.com
    TLS         on
    compress    gzip
    apikey     *****
    dd_service  ${DD_SERVICE}
    dd_source   Windows-events
    dd_tags     env:${DD_ENV}
``` 
* Environment name and version (e.g. Kubernetes? What version?): ECS EC2 
* Server type and version: Docker Containers
* Operating System and version: Windows Server Core 2019
* Filters and plugins:

**Additional context**
We are using fluent bit to parse & redirect logs from our containerised applications to Datadog. We observed a frequent fleunt-bit crash in our production setup where the log inflow is huge. We didn't see this behaviour in lower (dev/test) environments.

We also tried upgrading the fluent-bit version to v2.1.8 but after the upgrade it started crashing in lower regions too and more frequently.

**Comment by lecaros** [comment]
Hello @Eishi2012,
Can you please try with Fluent Bit 2.0 series?
We are investigating a similar issue, and it seems that 2.0 behaves better.

**Comment by Eishi2012** [comment]
We are running this on containerized setup. 

@lecaros I tried running 2.0.9 version of the fluent-bit app too but it did not solve the problem. The app keeps restarting every minute now.

`Fault bucket , type 0
Event Name: APPCRASH
Response: Not available
Cab Id: 0

Problem signature:
P1: fluent-bit.exe
P2: 2.0.9.0
P3: 63e109cb
P4: fluent-bit.exe
P5: 2.0.9.0
P6: 63e109cb
P7: c0000005
P8: 000000000011eb24
P9: 
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERA260.tmp.dmp
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERA38A.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERA485.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERA483.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WERA4A3.tmp.txt
\\?\C:\Windows\Temp\WERA4D5.tmp.appcompat.txt
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportQueue\AppCrash_fluent-bit.exe_67bf9ec6d4b36daccbf05e049bd79d6da927d93_56e17d0f_cab_1ac0a4f0\memory.hdmp

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportQueue\AppCrash_fluent-bit.exe_67bf9ec6d4b36daccbf05e049bd79d6da927d93_56e17d0f_cab_1ac0a4f0

Analysis symbol: 
Rechecking for solution: 0
Report Id: 367e7749-84b1-4680-8c41-9d40b77630e6
Report Status: 6
Hashed bucket: 
Cab Guid: 0`

**Comment by braydonk** [comment]
We are able to consistently reproduce this issue as well in my team's setup. We'll be investigating in the coming days.

**Comment by braydonk** [comment]
Pretty sure this was closed by #8179. Please re-open if that is not the case.
